### PR TITLE
OV-3017

### PR DESCRIPTION
**Pull request by OrigamiShiro**
add canvas group in prefab for castle screen buttons
add build logic
add animation for show builded building in castle screen
add draft async animation for showing castle screen buttons
add background music for DialogScreen
add possibility to hide / show game object with canvas group
add transition from battle4 to castle screen

### CODE REVIEW

File: Assets/Scripts/UI/Screens/MunicipalityScreen/MunicipalityScreen.cs [hunk 1]
```diff
@@ -198,6 +198,20 @@ private void Customize()
                 if (!buildingData.isMax)
                     MaxLevelByKey(buildingData.key).SetActive(false);
             }
+
+            if (inputData != null && inputData.isFTUEMode)
+            {
+                var chapter = GameData.GetFTUEChapterById(inputData.ftueStageData.ftueChapterData.id).key;
+                var lastStageKey = GameData.ftueStats.lastEndedStageData.key;
```
Comment (kbb1025doge) [inline Assets/Scripts/UI/Screens/MunicipalityScreen/MunicipalityScreen.cs:205]: inputData.ftueStageData.ftueChapterData.key

File: Assets/Scripts/UI/Screens/MunicipalityScreen/MunicipalityScreen.cs [hunk 2]
```diff
@@ -267,37 +300,74 @@ private void PortalButtonClick()
...
-                    mode = inputData?.mode
+                    ftueStageId = inputData.ftueStageId,
+                    mode = inputData.mode,
                 }).RunShowScreenProcess();
+            }
+            else
+            {
+                UIManager.ShowScreen<CastleScreen>();
+            }
         }
```
Comment (kbb1025doge) [inline Assets/Scripts/UI/Screens/MunicipalityScreen/MunicipalityScreen.cs:367]: switch

File: Assets/Scripts/UI/Screens/CastleScreen/BaseButton.cs [hunk 3]
```diff
@@ -16,6 +17,7 @@ public abstract class BaseButton : MonoBehaviour
             protected TextMeshProUGUI title;
 
             protected TextMeshProUGUI markers;
+            public int? stageId;
```
Comment (kbb1025doge) [inline Assets/Scripts/UI/Screens/CastleScreen/BaseButton.cs:20]: swap to mode

File: Assets/Scripts/UI/Screens/CastleScreen/MunicipalityButton.cs [hunk 4]
```diff
@@ -24,9 +25,26 @@ protected override void Awake()
...
+                                UIManager.ShowScreen<MunicipalityScreen>();
+                                break;
+                        }
+                        break;
+                    default:
+                        UIManager.ShowScreen<MunicipalityScreen>();
+                        break;
+                }
+            }
+
```
Comment (kbb1025doge) [inline Assets/Scripts/UI/Screens/CastleScreen/MunicipalityButton.cs:47]: судя по всему все пути ведут в MunicipalityScreen... посему надо быть проще ->
UIManager.ShowScreen<MunicipalityScreen>();

Comment (OrigamiShiro) [reply Assets/Scripts/UI/Screens/CastleScreen/MunicipalityButton.cs]: переделал :_)

**Pull request closed without merging**